scripts/netlifyPackage.js

```diff
@@ -0,0 +1,54 @@
+// @ts-check
+import assert from 'node:assert'
+import { basename, dirname, resolve } from 'node:path'
+import { argv } from 'node:process'
+import { fileURLToPath } from 'node:url'
+import { readFile, writeFile } from 'node:fs/promises'
+
+import execa from 'execa'
+
+const packageJSON = await getPackageJSON()
+
+async function getPackageJSON() {
+  const packageJSONPath = resolve(fileURLToPath(import.meta.url), '../../package.json')
+  const contents = JSON.parse(await readFile(packageJSONPath, 'utf8'))
+
+  return {
+    contents,
+    path: packageJSONPath,
+  }
+}
+
+/**
+ * @type {Record<string, function>}
+ */
+const commands = {
+  prepare: async () => {
+    const newPackageJSON = {
+      ...packageJSON.contents,
+      main: './dist/index.js',
+      name: 'netlify',
```

Comment:
Now I'm wondering if we should also change the name of the binary so that `npx netlify` and any globally-installed `netlify-cli` do not collide?

Reply:
As far as I understand the wonky npm heuristics here:
- if a user has both `netlify` and `netlify-cli` installed (both locally or both globally), `npx netlify` will run `netlify`
- if a user has `netlify-cli` and does not have `netlify`, `npx netlify` will always run `netlify-cli` rather than try to install `netlify`
- if a user has neither installed, `npx netlify` will install and run `netlify`

(It's likely worth verifying these)

Does this seem ok or are any of these scenarios problematic?

Reply:
My question is whether `npx netlify` will also install the binary into the PATH, in the same way that globally-installing a package does. If that's the case, then the `netlify` binary will overwrite the binary with the same name that had been injected by a globally-installed `netlify-cli`, which is not intended.

But if `npx netlify` simply runs the binary without adding it to the PATH (which I feel is the expected behaviour), this should be fine.